bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TableUtils.java

```diff
@@ -93,6 +94,8 @@ public static Long hash(Object obj, Node parent) {
                     result = 31 * result + hash(tableIterator.next(), node);
                 }
                 return result;
+            } else if (refValue instanceof RegExpValue) {
+                return (long) refValue.toString().hashCode();
```

Comment:
Can this line cause hash collision with a similar string value?

Reply:
No. Added tests to cover this.